src/io/syncsocket.c

```diff
@@ -393,6 +394,10 @@ static const MVMIOOps op_table = {
     gc_free
 };
 
+/* since we'll throw an error in case of an error we can turn off -Wreturn-type here,
+works with GCC and clang, others will ignore this */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreturn-type"
```

Comment:
We already have a standard way to do this kind of thing, which is to annotate the `throw_error` function with `MVM_NO_RETURN`. I think that would solve it too, and we already know that works across plenty of different compilers.